data-pipeline-ffi/src/error.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use data_pipeline::trace_exporter::error::{
+    BuilderErrorKind, NetworkErrorKind, TraceExporterError,
+};
+use std::ffi::{c_char, CString};
+use std::fmt::Display;
+use std::io::ErrorKind as IoErrorKind;
+
+/// Represent error codes that `Error` struct can hold
+#[repr(C)]
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum ExporterErrorCode {
+    AddressInUse,
+    ConnectionAborted,
+    ConnectionRefused,
+    ConnectionReset,
+    HttpBodyFormat,
+    HttpBodyTooLong,
+    HttpClient,
+    HttpParse,
+    HttpServer,
+    HttpWrongStatus,
+    InvalidArgument,
+    InvalidData,
+    InvalidInput,
+    InvalidUrl,
+    IoError,
+    NetworkUnknown,
+    Serde,
+    TimedOut,
+}
+
+impl Display for ExporterErrorCode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::AddressInUse => write!(f, "Address already in use"),
+            Self::ConnectionAborted => write!(f, "Connection aborted"),
+            Self::ConnectionRefused => write!(f, "Connection refused"),
+            Self::ConnectionReset => write!(f, "Connection reset by peer"),
+            Self::HttpBodyFormat => write!(f, "Error parsing HTTP body"),
+            Self::HttpBodyTooLong => write!(f, "HTTP body too long"),
+            Self::HttpClient => write!(f, "HTTP error orgininated by client"),
+            Self::HttpParse => write!(f, "Error while parsing HTTP message"),
+            Self::HttpServer => write!(f, "HTTP error orgininated by server"),
+            Self::HttpWrongStatus => write!(f, "HTTP wrong status number"),
+            Self::InvalidArgument => write!(f, "Invalid argument provided"),
+            Self::InvalidData => write!(f, "Invalid data payload"),
+            Self::InvalidInput => write!(f, "Invalid input"),
+            Self::InvalidUrl => write!(f, "Invalid URL"),
+            Self::IoError => write!(f, "Input/Output error"),
+            Self::NetworkUnknown => write!(f, "Unknown network error"),
+            Self::Serde => write!(f, "Serialization/Deserialization error"),
+            Self::TimedOut => write!(f, "Operation timed out"),
+        }
+    }
+}
+
+/// Stucture that contains error information that `TraceExporter` API can return.
+#[repr(C)]
+#[derive(Debug, PartialEq)]
+pub struct ExporterError {
```

Comment:
There are already error types in ddcommon.  Maybe either use those, or move this there?

Reply:
Hey @danielsn made a mistake with this PR, I accidentally duplicated it. If you have further comments please do it in this [one](https://github.com/DataDog/libdatadog/pull/768). Sorry for the inconvenience.
Regarding your question, we wanted to move forward quickly so we decided to handle errors internally and later, in a separate PR, try to port this to the common module improving the current `MaybeError` data type.